Lib/test/test_dis.py

```diff
@@ -1350,8 +1376,9 @@ async def async_def():
    0: None
    1: 1
 Names:
-   0: b
-   1: c
+   0: value
+   1: b
+   2: c
 Variable names:
    0: a
    1: d"""
```

Comment:
It would be good to also add a test that shows what you're fixing here.

Reply:
Hm, we're not really fixing anything (in fact, we want the behavior to stay exactly the same). All we're really doing is moving some of the error handling logic for `close` and `throw` into the bytecode, so I'm not really sure how to test that.

I'm confident that the tests in `test_yield_from.TestInterestingEdgeCases` exercise this path enough to prevent any behavior changes (like we've seen with other approaches to address this issue).